packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts

```diff
@@ -15,6 +15,15 @@ const spinButtonExtraClassNames = {
   buttonActive: 'fui-SpinButton__button_active',
 };
 
+const appearanceMap: Record<string, string> = {
```

Comment:
You can type this as the following to avoid the casts below

```suggestion
const appearanceMap: Record<string, keyof typeof buttonStyles> = {
```

Reply:
Maybe even?

```
const appearanceMap: Record<SpinButtonProps['appearance'], keyof typeof buttonStyles> = {
```

Reply:
we can't since we have these cases:

`buttonDisabledStyles[appearanceMap[appearance] as keyof typeof buttonDisabledStyles]`
`buttonStyles[appearanceMap[appearance] as keyof typeof buttonStyles]`

Reply:
The other thing you could do is move this object's creation within the `useSpinButtonStyles_unstable` function after the creation of `buttonStyles` (maybe around line 405). Then you could type it as I said in the comment above and remove the casts. If you don't want the object to be generated on every render you can use `useMemo` with an empty array as the dependency list

```ts
const appearanceMap: Record<SpinButtonProps['appearance'], keyof typeof buttonStyles> = React.useMemo(() => ({
  outline: 'outline',
  underline: 'underline',
  'filled-darker': 'filled-darker',
  'filled-lighter': 'filled-lighter',
  'filled-darker-shadow': 'filled-darker',
  'filled-lighter-shadow': 'filled-lighter',
}), [])
```